src/data/battler-tags.ts

```diff
@@ -2109,6 +2114,48 @@ export class CritBoostTag extends BattlerTag {
   }
 }
 
+/**
+ * Used for an instance of a stackable crit boost
+ * Granted by G-Max Chi Strike
+ */
+export class ChiStrikeCritBoostTag extends BattlerTag {
+  public stackCount: number = 0;
+
+  constructor(tagType: BattlerTagType, sourceMove: Moves) {
+    super(tagType, BattlerTagLapseType.TURN_END, 1, sourceMove, undefined, true);
+  }
+
+  override onAdd(pokemon: Pokemon): void {
+    super.onAdd(pokemon);
+    this.stackCount += 1;
+
+    globalScene.queueMessage(
+      i18next.t("battlerTags:chiStrikeCritBoostOnAdd", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }),
+    );
+  }
+
+  override lapse(pokemon: Pokemon, lapseType: BattlerTagLapseType): boolean {
+    return lapseType !== BattlerTagLapseType.CUSTOM || super.lapse(pokemon, lapseType);
+  }
+
+  override onOverlap(pokemon: Pokemon): void {
+    this.onAdd(pokemon);
+  }
+
+  override onRemove(pokemon: Pokemon): void {
+    super.onRemove(pokemon);
+
+    globalScene.queueMessage(
+      i18next.t("battlerTags:chiStrikeCritBoostOnRemove", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }),
+    );
+  }
+
+  override loadTag(source: BattlerTag | any): void {
+    super.loadTag(source);
+    this.stackCount = source.stackCount || 0;
```

Comment:
Always `??` over `||` unless you are 100% sure you want `||`

```suggestion
    this.stackCount = source.stackCount ?? 0;
```

Reply:
I stole that code from stockpile, I can add that over there too then